packages/base-map/src/index.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { MapProps, MapRef, Map, MapProvider } from "react-map-gl";
+import maplibregl, { Event } from "maplibre-gl";
+
+import * as Styled from "./styled";
+import callIfValid from "./util";
+import MarkerWithPopup from "./MarkerWithPopup";
+
+/**
+ * The BaseMap component renders a MapLibre map
+ * markers that are declared as child elements of the BaseMap element.
+ *
+ * As BaseMap wraps a react-map-gl Map component, ny control which can be added as a child of a react-map-gl map is supported.
+ * See https://visgl.github.io/react-map-gl/docs/api-reference/map to see which react-map-gl
+ * children are shipped by default. Others are also supported.
+ *
+ * Overlays are groups of similar MapLibre markers, e.g. vehicle location
+ * markers, bus stop markers, etc.
+ *
+ * Overlays are automatically added to the overlay control displayed by the
+ * BaseMap. The user uses that control to turn overlays on or off. Only overlays
+ * with an id are added to the control.
+ */
+type Props = React.ComponentPropsWithoutRef<React.ElementType> & {
+  baseLayer?: string;
+  center?: [number, number];
+  forceMaxHeight?: boolean;
+  mapLibreProps?: MapProps;
+  maxZoom?: number;
+  onClick?: (evt: Event) => void;
+  // Unknown is used here because of a maplibre/mapbox issue with the true type, MapLayerMouseEvent
+  onContextMenu?: (e: unknown) => void;
+  // TODO: does this cause integration issues?
+  onViewportChanged?: (e: maplibregl.MapLibreEvent) => void;
+  passedRef?: React.Ref<MapRef>;
+  zoom?: number;
+};
+type State = {
+  latitude: number;
+  longitude: number;
+  zoom: number;
+  fitBoundsOptions?: Record<string, number | string | boolean>;
+};
+
+const BaseMap = ({
+  // These tiles are free to use, but not in production
+  baseLayer = "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
+  center,
+  children,
+  forceMaxHeight,
+  mapLibreProps,
+  maxZoom,
+  onClick,
+  onContextMenu,
+  passedRef,
+  onViewportChanged,
+  zoom: initZoom = 12
+}: Props): JSX.Element => {
+  const [viewState, setViewState] = React.useState<State>({
+    fitBoundsOptions: {
+      animate: true,
+      duration: 300,
+      essential: false,
+      maxDuration: 600,
+      padding: 200
+    },
```

Comment:
Is `fitBoundsOptions` used anywhere? It seems like event `onMove` is the only place where `setViewState` is passed props other than latitude and longitude, and that event does not pass fitting boundaries. `fitBoundsOptions` gets passed around via `onViewportChanged`, but the content seems irrelevant.

Reply:
These options would normally be passed in via props to MapLibreGL, but since we're using a controlled component we must set these in state. It's not possible to have some elements of viewState be controlled and some uncontrolled.